rom/dev/src/flow/cold_reset/idev_id.rs

```diff
@@ -185,16 +208,22 @@ impl InitDevIdLayer {
     fn derive_key_pair(
         env: &mut RomEnv,
         cdi: KeyId,
-        priv_key: KeyId,
-    ) -> CaliptraResult<Ecc384KeyPair> {
-        let result = Crypto::ecc384_key_gen(env, cdi, b"idevid_keygen", priv_key);
+        ecdsa_priv_key: KeyId,
+        mldsa_keypair_seed: KeyId,
+    ) -> CaliptraResult<(Ecc384KeyPair, MlDsa87PubKey)> {
+        let result = Crypto::ecc384_key_gen(env, cdi, b"idevid_ecc_key", ecdsa_priv_key);
         if cfi_launder(result.is_ok()) {
             cfi_assert!(result.is_ok());
             report_boot_status(IDevIdKeyPairDerivationComplete.into());
         } else {
             cfi_assert!(result.is_err());
         }
-        result
+        let ecc_keypair = result?;
+
+        // Derive the MLDSA Key Pair.
+        let mldsa_pub_key =
```

Comment:
For consistency we should move the ML-DSA keygen to be before `report_boot_status(IDevIdKeyPairDerivationComplete.into())`, since IDevID keypair derivation is not complete until we've done both.

Reply:
Updated.